### ISSUE TITLE
Error: It is not possible to add a breakpoint at the location requested.

### ISSUE THREAD

**Issue by NemanjaKutlesic** [microsoft/AL#5597]
**Describe the bug**
I'm not able to set breakpoints in my al code

**To Reproduce**
Im running the latest version on docker, and on some places i can set breakpoint, on some i cannot and i got the error from title

**Expected behavior**
I should be able to set breakpoints everywhere

**Screenshots**
![image](https://user-images.githubusercontent.com/54843470/72159340-6c2d5580-33bc-11ea-8a99-f753a2b6a72f.png)

**Version:**

 - AL Language: 4.0.206840

**Comment by LFlamme** [comment]
same problem. Restarting VSCode fixes it for me :)

**Comment by NemanjaKutlesic** [comment]
It is not temporary solution, some of them work after restart, but some still remains dead

**Comment by mjmatthiesen** [comment]
Same problem. Came with a recent update. Same as begin/end autocomplete being entirely broken now.

**Comment by kalberes** [comment]
Can you provide a repro.?

**Comment by daansaveyn** [comment]
This issue also occurs quite frequently on my dev environment. I could not figure out the exact situation until a couple of minutes ago.

In my case the following scenario creates the issue:
- Change a piece of code
- Do a RAD
- Select "Debug without publishing" from VSCode and wait untill the session is started.
- Try to add a breakpoint withing the code you've just changed.

VSCode now throws the error "Error: It is not possible to add a breakpoint at the location requested.". 

This leads me to believe it has something to do with Rapid Application Publishing.

**Comment by LFlamme** [comment]
Well if you use "Debug without publishing" and you changed the code it actually makes sense if it doesn´t work, as the debugger will debug code thats different compared to yours. 
If you for exaple add a new Line (5) with a command the debugger would show you your new Line but it would actually be 1 line further...

**Comment by kalberes** [comment]
@daansaveyn 

1. What does it mean "Do a RAD"? Do an Alt Ctrl F5? RAd publishing without debugging?
2. Have you tried Alt F5 publishing with debugging. Does that work?

I have no repro on this on the latest build(for you insider docker) I have assumed "Do a RAD" being an  Alt Ctrl F5. But there were no changes in RAD compared to the released version 15.x

**Comment by mjmatthiesen** [comment]
Here's an easy example:
```
        case precrRecRef.Number() of
            Database::"Warehouse Shipment Header":
                begin
                    precrRecRef.SetTable(lrecWhseShipHeader);
                end;
            Database::"Transfer Header":
                begin
                    precrRecRef.SetTable(lrecTransferHeader);
                end;
            Database::"Sales Header":
                begin
                    precrRecRef.SetTable(lrecSalesHeader);
                end;
        end;
```
You cannot set a breakpoint on case. It will be valid, until the code runs and deploys, then the breakpoint is disabled. This is annoying when I want to examine the case statement specifically, because I need a break statement in each case instead.

**Comment by bleisibach** [comment]
I can confirm mjmatthiesen: we can not debug case structures.

**Comment by rznd** [comment]
I cannot debug this case statement on version 16 RTM , using Docker container:

![image](https://user-images.githubusercontent.com/24936737/81197237-6e807480-8f96-11ea-8482-98fc4733a8bf.png)

**Comment by achecac** [comment]
Neither do I. I cannot debug case structures
Versión: ES Business Central 16.0 (Plataforma 16.0.11233.13086 + Aplicación 16.0.11240.11946)

**Comment by guilliano** [comment]
I have the same problem:
![image](https://user-images.githubusercontent.com/4299867/83744645-b7dae700-a65c-11ea-8569-87b97b8efd94.png)

**Comment by kalberes** [comment]
I accept it as a bug for now.

**Comment by atoader** [comment]
The issue has been fixed and checked into the master branch. It will be available in the next major release of Business Central or through the preview program. Please try it out and let us know if you still encounter the issue. As requested through the support channel, the fix will also be backported to 16.x .

**Comment by albgen** [comment]
hello,
i have a similar issue. I can set breakpoint in some of the methods of sales line.dal but on some of them cannot.
I thought problems with symblos and cache so i redownloaded them but the issue still exist.
Is this fix related to this kind of issue?

**Comment by thloke** [comment]
Issue has been fixed since 16.6
